chart/templates/configMap.yaml

```diff
@@ -11,3 +11,6 @@ data:
   {{- else if .Values.webhook }}
   rancher-webhook: {{ toYaml .Values.webhook | quote }}
   {{- end }}
+  {{- with .Values.fleet }}
+  fleet: {{ kindIs "string" . | ternary (. | quote) (toYaml . | quote) }}
```

Comment:
Is there a reason not to use the same logic as above? With the above approach, users can set helm values from a file `--set-file outer.inner=value.yaml`  or use the basic parameter `--set outer.inner=value`.